mne/tests/test_dipole.py

```diff
@@ -151,3 +157,80 @@ def test_len_index_dipoles():
 
 
 run_tests_if_main(False)
+
+
+@requires_sklearn
+@testing.requires_testing_data
+def test_min_distance_fit_dipole():
+
+    data_path = testing.data_path()
+    raw_fname = data_path + '/MEG/sample/sample_audvis_trunc_raw.fif'
+
+    subjects_dir = op.join(data_path, 'subjects')
+    fname_cov = op.join(data_path, 'MEG', 'sample', 'sample_audvis-cov.fif')
+    fname_trans = op.join(data_path, 'MEG', 'sample',
+                          'sample_audvis_trunc-trans.fif')
+    fname_bem = op.join(subjects_dir, 'sample', 'bem',
+                        'sample-1280-1280-1280-bem-sol.fif')
+
+    subject = 'sample'
+
+    raw = Raw(raw_fname, preload=True)
+
+    # select eeg data
+    picks = pick_types(raw.info, meg=False, eeg=True, exclude='bads')
+    info = pick_info(raw.info, picks)
+
+    # Let's use cov = Identity
+    cov = read_cov(fname_cov)
+    cov['data'] = np.eye(cov['data'].shape[0])
+
+    # Simulated scal map
+    simulated_scalp_map = np.zeros(picks.shape[0])
+    simulated_scalp_map[27] = 1
+    simulated_scalp_map[28] = 1
+    simulated_scalp_map[29] = 1
+    simulated_scalp_map[30] = 1
+    simulated_scalp_map[31] = 1
+    simulated_scalp_map[32] = 1
+    simulated_scalp_map[33] = 1
+
+    simulated_scalp_map = simulated_scalp_map[:, None]
+
+    evoked = EvokedArray(simulated_scalp_map, info, tmin=0)
+
+    min_distance = 5  # distance in mm
+
+    dip, residual = fit_dipole(evoked, cov, fname_bem, fname_trans,
+                               min_dist=min_distance)
+
+    dist, radiality = _compute_depth_and_radiality(dip, fname_trans,
+                                                   subject, subjects_dir)
+
+    assert (dist * 1000) > min_distance
+
+
+def _compute_depth_and_radiality(dip, fname_trans, subject, subjects_dir):
+    trans = read_trans(fname_trans)
+    trans = _get_mri_head_t(trans)[0]
+    subjects_dir = get_subjects_dir(subjects_dir=subjects_dir)
+    fname = op.join(subjects_dir, subject, 'bem',
+                    'inner_skull.surf')
+    points, faces = read_surface(fname)
+    points = apply_trans(trans['trans'], points * 1e-3)
+
+    pos = dip.pos
+    ori = dip.ori
+
+    from sklearn.neighbors import NearestNeighbors
+    nn = NearestNeighbors()
+    nn.fit(points)
+    depth, idx = nn.kneighbors(pos, 1, return_distance=True)
```

Comment:
For the dipole fit, you mean? that only uses `scipy`. I thought we had a function that looked for nearest neighbors already using either `scipy` or `sklearn` depending on user capability, but maybe we don't. It seems like it could just be a `np.min` of a `cdist`-like operation (which we definitely have somewhere) even if we don't have a nearest-neighbor function, though.